doc/sacred.md

```diff
@@ -0,0 +1,175 @@
+[the config](https://sacred.readthedocs.io/en/stable/configuration.html) and 
+[the command line interface](https://sacred.readthedocs.io/en/stable/command_line.html).
+ 
+Consider the following example for a trainings script using sacred:
+
+```python
+from sacred import Experiment
+from sacred.observers import FileStorageObserver
+import padertorch as pt
+from pathlib import Path
+
+# Define the experiment. Experiment is the main class for managing experiments.
+ex = Experiment('my-experiment-name')
+
+
+class MyModel(pt.Model):
+    # Implement your model here
+    def __init__(self, num_layers=2):
+        super().__init__()
+        self.num_layers = num_layers
+
+
+# Create a ConfigScope with the config decorator
+@ex.config
+def config():
+    """
+    This function contains the configuration of the experiment. All local 
+    variables in this function will be captured by sacred and used as 
+    configuration values.
+    """
+    # padertorch.Trainer inherits from Configurable, so we can easily construct 
+    # the config dictionary here and instantiate the trainer from the config 
+    # later
+    trainer = {
+        # The 'factory' key tells the Configurable which class to instantiate
+        'factory': pt.Trainer,
+        # The next line creates a new experiment folder for this experiment
+        'storage_dir': pt.io.get_new_storage_dir(ex.path),
+        'model': {
+            'factory': MyModel, # <-- insert your model class here
+            # You can put additional configuration of your model here
+        },
+        # Here goes additional configuration of the trainer
+    }   
+    
+    # The following line fills the trainer config with the default arguments of
+    # the configurable classes. In this example, it will, for example, insert 
+    # the "num_layers" argument from the model.
+    pt.Trainer.get_config(trainer)
+    
+    # This adds an observer to the experiment that stores the collected 
+    # information about the experiment. In this case, we use a 
+    # FileStorageObserver that writes the information to the filesystem to a 
+    # sub-folder of the experiment's storage directory.
+    ex.observers.append(FileStorageObserver(
+        Path(trainer['storage_dir']) / 'sacred')
+    )
+
+
+@ex.automain
+def main(trainer, _config):
+    """This is the main function of your experiment. It receives all 
+    configuration values set in `config` as parameters. `_config` contains the  
+    whole configuration as a dictionary."""
+
+    # The following line creates the trainer from the configuration using 
+    # classmethods provided by the Configurable class
+    trainer = pt.Trainer.from_config(trainer)
+
+    # This stores the configuration in the way it is expected by the model 
+    # loading functions in padertorch (pt.Model.from_storage_dir)
+    pt.io.dump_config(_config, trainer.storage_dir / 'config.json')
```

Comment:
I think for configs it is better to use yaml.
It is better for humans to read and edit.
The drawback, that it is slower, doesn't matter for so small files.

Reply:
I disagree, we should only promote something we are also using. Most if not all our examples are using json so that it would look strange if we promote something here which we do not heed.

Reply:
And also the default value for `pt.Model.from_storage_dir` is `config.json`, not yaml. We wold have to change that if we recommend to use yaml (or infer the file type)

Reply:
Good point.
I think using the json comes from the fact, that sacred has json as default and originally, sacred was responsible to write the config.
Maybe I should open an issue and start a discussion, if we should change to yaml as default.